prow/config/config.go

```diff
@@ -109,6 +109,9 @@ type ProwConfig struct {
 	// PushGateway is a prometheus push gateway.
 	PushGateway PushGateway `json:"push_gateway,omitempty"`
 
+	// ServeMetrics tells if or not the components serve metrics
+	ServeMetrics bool `json:"serve_metrics"`
```

Comment:
Nit: can this be under PushGateway?

Reply:
thought about this too.
Apparently, with the logic you described below, it makes sense to put it under.
With my original logic, they are independent of each other.

Reply:
Yeah we want the logic I described since it's purely additive to the current behavior and does not change the current behavior unless it's opt-in